cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -29,50 +31,122 @@
 public abstract class RowColumns<T> {
   private final byte[] row;
 
-  private final List<byte[]> columns;
-
-  public byte[] getRow() {
-    return row;
-  }
-
-  public List<byte[]> getColumns() {
-    return columns;
-  }
+  private List<byte[]> columns;
 
   // key & column as byte[]
 
+  /**
+   * Defines a row, including all columns, unless additional columns are specified.
+   * @param row the row to define.
+   */
   public RowColumns(byte[] row) {
     this.row = row;
-    this.columns = new ArrayList<>();
   }
 
+  /**
+   * Defines a row, with the specified set of columns.
+   * @param row the row to define
+   * @param columns the set of columns of the row to define
+   */
   public RowColumns(byte[] row, byte[]... columns) {
-    this(row);
-    add(columns);
-  }
-
-  @SuppressWarnings("unchecked")
-  public T add(byte[]... columns) {
-    Collections.addAll(this.columns, columns);
-    return (T) this;
+    this.row = row;
+    this.columns = Arrays.asList(columns);
   }
 
   // key and column as String
 
+  /**
+   * Defines a row, including all columns, unless additional columns are specified.
```

Comment:
Same as above. Also good to specific the provided String row key is converted to `byte[]` with UTF-8 encoding.